std/algorithm/iteration.d

```diff
@@ -1103,7 +1104,11 @@ if (is(typeof(unaryFun!predicate)))
 {
     auto filter(Range)(Range range) if (isInputRange!(Unqual!Range))
     {
-        return FilterResult!(unaryFun!predicate, Range)(range);
+        import std.range : SortedRange, assumeSorted;
+        static if (is(Range : SortedRange!TT, TT))
+            return assumeSorted(FilterResult!(unaryFun!predicate, Range)(range));
```

Comment:
The sorting predicate should be propagated here, e.g. `return FilterResult!(unaryFun!predicate, Range)(range).assumeSorted!TT;`